### PR TITLE
[ci.yaml] Add platform_properties, add_recipe_cq

### PR DESCRIPTION
**Pull request by CaseyHillers**
https://github.com/flutter/flutter/issues/84998

Related infra CL: https://flutter-review.googlesource.com/c/infra/+/15360

### CODE REVIEW

File: .ci.yaml [hunk 1]
```diff
@@ -1,17 +1,40 @@
...
+# for every commit.
+#
+# More information at:
+#  * https://github.com/flutter/cocoon/blob/master/CI_YAML.md
 enabled_branches:
   - master
 
+platform_properties:
+  linux:
+    properties:
```
Comment (christopherfujino) [inline .ci.yaml:13]: do we need a `properties` key here? are there gonna be other keys at this same level?

Comment (christopherfujino) [reply .ci.yaml:13]: nevermind, I see this is already defined in the proto

Comment (CaseyHillers) [reply .ci.yaml:13]: Unfortunately, yes. I can't find a way around it as protos do not make it possible to nest values (in this case we map->properties->map)

We could hardcode the platforms in the proto (linux_properties->map), but i'd prefer to make it easy to add new platforms (like web, linux_android)

Comment (christopherfujino) [reply .ci.yaml:13]: agreed. this is annoying, but fine.

**Comment by christopherfujino** [comment]
Looks like a parse error:

```
00:00 [32m+2[0m[31m -1[0m: /b/s/w/ir/k/cocoon/app_dart/integration_test/validate_all_ci_configs_test.dart: validate config file of flutter/flutter/master/.ci.yaml [1m[31m[E][0m[0m
  Protobuf JSON decoding failed at: root["targets"]["0"]["tags"]["0"]. Expected String value
  package:test_api                                         fail
  integration_test/validate_all_ci_configs_test.dart 37:9  main.<fn>
```

File: .ci.yaml [hunk 2]
```diff
@@ -1,17 +1,40 @@
...
+      caches: "[{\"name\": \"dart_pub_cache\", \"path\": \".pub_cache\"}]"
```
Comment (CaseyHillers) [inline .ci.yaml:14]: @christopherfujino I updated this to be json encoded. For repos with more caches, we can use a yaml multiline block to make this prettier

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**